### PR TITLE
Added negative TCs for resume upload feature of zbox cli

### PR DESCRIPTION
**Pull request by Zorro30**
Negative testcase to test resume upload flow for zbox cli

### CODE REVIEW

**Review by Jayashsatolia403 — CHANGES_REQUESTED** [review]
Please check comments

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go [hunk 1]
```diff
@@ -80,4 +80,157 @@ func TestResumeUpload(testSetup *testing.T) {
 		)
 		require.Equal(t, expected, output[1])
 	})
+
+	t.RunSequentiallyWithTimeout("Resume upload with diff file name", 10*time.Minute, func(t *test.SystemTest) {
+		allocSize := int64(2 * GB)
+		fileSize := int64(500 * MB)
```
[the remark this answers is not in the document]
Comment (Zorro30) [reply tests/cli_tests/0_zboxcli_file_resume_upload_test.go:86]: done

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go [hunk 2]
```diff
@@ -80,4 +80,157 @@ func TestResumeUpload(testSetup *testing.T) {
...
+		allocationID := setupAllocation(t, configPath, map[string]interface{}{
+			"size": allocSize,
+			"lock": 50,
+		})
+
+		filename := generateRandomTestFileName(t)
+		filename2 := generateRandomTestFileName(t)
+		err = createFileWithSize(filename, fileSize)
+		require.Nil(t, err)
+		defer func() {
```
[the remark this answers is not in the document]
Comment (Zorro30) [reply tests/cli_tests/0_zboxcli_file_resume_upload_test.go:100]: These delete doesn't affect the flow right I am anyways creating a new file to upload in the second run right.

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go [hunk 3]
```diff
@@ -80,4 +80,157 @@ func TestResumeUpload(testSetup *testing.T) {
...
+			"./zbox upload %s --silent --wallet %s --configDir ./config --config %s",
+			upload_param,
+			escapedTestName(t)+"_wallet.json",
+			configPath,
+		)
+
+		cmd, _ := cliutils.StartCommandWithoutRetry(command)
+		uploaded := waitPartialUploadAndInterrupt(t, cmd)
+		t.Logf("the uploaded is %v ", uploaded)
+
```
[the remark this answers is not in the document]
Comment (Zorro30) [reply tests/cli_tests/0_zboxcli_file_resume_upload_test.go:121]: I have done the same I guess!

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go [hunk 4]
```diff
@@ -80,4 +80,157 @@ func TestResumeUpload(testSetup *testing.T) {
...
+			"allocation":  allocationID,
+			"remotepath":  "/demo/",
+			"localpath":   filename,
+			"chunknumber": 500, // 64KB * 500 = 32M
+		}, false)
+
+		require.Nil(t, err, strings.Join(output, "\n"))
+	})
+
+	t.RunSequentiallyWithTimeout("Resume upload with diff allocationID", 10*time.Minute, func(t *test.SystemTest) {
```
[the remark this answers is not in the document]
Comment (Zorro30) [reply tests/cli_tests/0_zboxcli_file_resume_upload_test.go:183]: okay

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go [hunk 5]
```diff
@@ -80,4 +80,157 @@ func TestResumeUpload(testSetup *testing.T) {
...
+			escapedTestName(t)+"_wallet.json",
+			configPath,
+		)
+
+		cmd, _ := cliutils.StartCommandWithoutRetry(command)
+		uploaded := waitPartialUploadAndInterrupt(t, cmd)
+		t.Logf("the uploaded is %v ", uploaded)
+
+		// increasing the file size to test the negative flow
+		err = createFileWithSize(filename, fileSize2)
```
[the remark this answers is not in the document]
Comment (Zorro30) [reply tests/cli_tests/0_zboxcli_file_resume_upload_test.go:172]: We already have a positive case where we have same size scenario, hence only added a case for different szie.

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go [hunk 6]
```diff
@@ -80,4 +80,157 @@ func TestResumeUpload(testSetup *testing.T) {
...
+		output, err = uploadFile(t, configPath, map[string]interface{}{
+			"allocation":  allocationID,
+			"remotepath":  "/demo/",
```
[the remark this answers is not in the document]
Comment (Zorro30) [reply tests/cli_tests/0_zboxcli_file_resume_upload_test.go:175]: I tried that, but even that was successful.

[the remark this answers is not in the document]
Comment (Zorro30) [reply tests/cli_tests/0_zboxcli_file_resume_upload_test.go:175]: I added the 
```
remotepath : "/"
"localpath":   filename, #this has a different file size
```
still the testcase succeeded.

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go [hunk 7]
```diff
@@ -80,4 +80,157 @@ func TestResumeUpload(testSetup *testing.T) {
...
+		t.Logf("the uploaded is %v ", uploaded)
+
+		output, err = uploadFile(t, configPath, map[string]interface{}{
+			"allocation":  allocationID,
+			"remotepath":  "/",
+			"localpath":   filename2,
+			"chunknumber": 500, // 64KB * 500 = 32M
+		}, false)
+
+		require.NotNil(t, err, strings.Join(output, "\n"))
```
[the remark this answers is not in the document]
Comment (Zorro30) [reply tests/cli_tests/0_zboxcli_file_resume_upload_test.go:129]: updated!

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (Jayashsatolia403) [reply tests/cli_tests/0_zboxcli_file_resume_upload_test.go:179]: Both are negative cases. same size different file and different size different file.

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (Jayashsatolia403) [reply tests/cli_tests/0_zboxcli_file_resume_upload_test.go]: If resume on same path is successful, that is a bug in system. Please report issues on channel if you notice something like this.

Please revert back to resume on same path code and share a successful run.

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go [hunk 8]
```diff
@@ -80,4 +80,157 @@ func TestResumeUpload(testSetup *testing.T) {
...
+		// increasing the file size to test the negative flow
+		err = createFileWithSize(filename, fileSize2)
+		output, err = uploadFile(t, configPath, map[string]interface{}{
+			"allocation":  allocationID,
+			"remotepath":  "/demo/",
+			"localpath":   filename,
+			"chunknumber": 500, // 64KB * 500 = 32M
+		}, false)
+
+		require.Nil(t, err, strings.Join(output, "\n"))
```
[the remark this answers is not in the document]
Comment (Zorro30) [reply tests/cli_tests/0_zboxcli_file_resume_upload_test.go]: done

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go (hunk 6 above)
[the remark this answers is not in the document]
Comment (Zorro30) [reply tests/cli_tests/0_zboxcli_file_resume_upload_test.go]: I have already reverted back to same path code & the system tests have passed for the PR.

**Review by Jayashsatolia403 — CHANGES_REQUESTED** [review]

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (Zorro30) [reply tests/cli_tests/0_zboxcli_file_resume_upload_test.go:100]: connected & resolved!

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (Zorro30) [reply tests/cli_tests/0_zboxcli_file_resume_upload_test.go:122]: okay, added the a dummy path in remote path variable.

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (Zorro30) [reply tests/cli_tests/0_zboxcli_file_resume_upload_test.go:180]: okay added a separate case for diff filename having diff file size

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go [hunk 9]
```diff
@@ -80,4 +80,185 @@ func TestResumeUpload(testSetup *testing.T) {
...
+		output, err := executeFaucetWithTokens(t, configPath, 100.0)
+		require.Nil(t, err, "error executing faucet", strings.Join(output, "\n"))
+
+		allocationID := setupAllocation(t, configPath, map[string]interface{}{
+			"size": allocSize,
+			"lock": 50,
+		})
+
+		filename := generateRandomTestFileName(t)
+		filename2 := generateRandomTestFileName(t)
```
Comment (Jayashsatolia403) [inline tests/cli_tests/0_zboxcli_file_resume_upload_test.go:97]: Rename same file

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go [hunk 10]
```diff
@@ -80,4 +80,185 @@ func TestResumeUpload(testSetup *testing.T) {
...
+		output, err = uploadFile(t, configPath, map[string]interface{}{
+			"allocation":  allocationID,
+			"remotepath":  "/dummy",
+			"localpath":   filename2,
+			"chunknumber": 500, // 64KB * 500 = 32M
+		}, false)
+
+		require.NotNil(t, err, strings.Join(output, "\n"))
+		//asserting output
+		require.Contains(t, output[0],"no such file or directory")
```
Comment (Jayashsatolia403) [inline tests/cli_tests/0_zboxcli_file_resume_upload_test.go:132]: Assert statement doesn't look correct.

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go [hunk 11]
```diff
@@ -80,4 +80,185 @@ func TestResumeUpload(testSetup *testing.T) {
...
+		require.Equal(t, expected, output[1])
+		//asserting error
+		require.Error(t, err)
+		expected = fmt.Sprintf(
+			"exit status 1",
+		)
+		require.Equal(t, expected, err.Error())
+	})
+
+	t.RunSequentiallyWithTimeout("Resume upload with diff filesize and diff filename (Negative)", 10*time.Minute, func(t *test.SystemTest) {
```
Comment (Jayashsatolia403) [inline tests/cli_tests/0_zboxcli_file_resume_upload_test.go:202]: It should be different file of same size.

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go [hunk 12]
```diff
@@ -80,4 +80,185 @@ func TestResumeUpload(testSetup *testing.T) {
...
+		require.Contains(t, output[0],"no such file or directory")
+		require.Error(t, err)
+		//asserting error
+		expected := fmt.Sprintf(
+			"exit status 1",
+		)
+		require.Equal(t, expected, err.Error())
+	})
+
+	t.RunSequentiallyWithTimeout("Resume upload with diff filesize having same filename (Negative)", 10*time.Minute, func(t *test.SystemTest) {
```
Comment (Jayashsatolia403) [inline tests/cli_tests/0_zboxcli_file_resume_upload_test.go:141]: It should be different file of different size.

**Review by Jayashsatolia403 — CHANGES_REQUESTED** [review]

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go (hunk 10 above)
Comment (zorro-30) [reply tests/cli_tests/0_zboxcli_file_resume_upload_test.go:132]: the file path coming in local & dev network is not the same, cannot generalise the assertions, hence asserting it with the constant thing what also comes in error.

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go (hunk 9 above)
Comment (zorro-30) [reply tests/cli_tests/0_zboxcli_file_resume_upload_test.go:97]: done

File: tests/cli_tests/0_zboxcli_file_resume_upload_test.go (hunk 12 above)
Comment (Zorro30) [reply tests/cli_tests/0_zboxcli_file_resume_upload_test.go]: Resolved

**Pull request merged**